boofuzz/socket_connection.py

```diff
@@ -241,6 +246,8 @@ def send(self, data):
         Returns:
             int: Number of bytes actually sent.
         """
+        num_sent = 0
```

Comment:
Just curious, were these changes (this and `data = b""` and `self._serverSock = None`) requested by Stickler or some other tool?

Reply:
Those were warnings in PyCharm.
Looking closer at the functions, all cases seem to be covered so that a reference before assignment should be impossible. PyCharm might not be smart enough to see this, but I guess it doesn't hurt to just initialise the return value. Just to be on the safe side and to appease PyCharm :D.

About `_serverSock`, it was initialised outside of the init function, with PyCharm complaining about this too.